packages/jaeger-ui/src/components/Monitor/ServicesView/serviceGraph.tsx

```diff
@@ -150,52 +150,68 @@ export class ServiceGraphImpl extends React.PureComponent<TProps> {
     const noDataComponent = this.generatePlaceholder('No Data');
     const apiErrorComponent = this.generatePlaceholder('Couldn’t fetch data');
 
-    const Plot = (
-      <XYPlot
-        margin={{ bottom: 25 }}
-        onMouseLeave={() => this.setState({ crosshairValues: [] })}
-        width={width}
-        height={this.height - 74}
-        yDomain={yDomain}
-      >
-        {showHorizontalLines ? <HorizontalGridLines /> : null}
-        <XAxis tickFormat={tickFormat} tickTotal={Math.floor(width / 60)} />
-        <YAxis />
-        {this.renderLines()}
-        <Crosshair values={this.state.crosshairValues}>
-          <div style={{ width: 140 }}>
-            {this.state.crosshairValues[0] &&
-              `${new Date(this.state.crosshairValues[0].x).toLocaleDateString()} ${new Date(
-                this.state.crosshairValues[0].x
-              ).toLocaleTimeString()}`}
-            {this.state.crosshairValues.reverse().map((d: TCrossHairValues) =>
-              showLegend ? (
-                <div key={d.label}>
-                  P{d.label * 100}: {d.y}
-                </div>
-              ) : (
-                <div key={d.label}>{d.y}</div>
-              )
-            )}
-          </div>
-        </Crosshair>
-        {showLegend ? (
-          <DiscreteColorLegend
-            className="legend-label"
-            orientation="horizontal"
-            items={this.getData()
-              .map((d: ServiceMetricsObject, idx: number) => ({
-                color: this.colors[idx],
-                title: `${d.quantile * 100}th`,
-              }))
-              .reverse()}
-          />
-        ) : null}
-      </XYPlot>
-    );
+    const Plot = () => {
+      let maxValue = 0;
+
+      if (metricsData && Array.isArray(metricsData)) {
+        const allMaxMetrics = metricsData.map(x => x.max);
+        maxValue = Math.max(...allMaxMetrics);
+      } else if (metricsData) {
+        maxValue = metricsData.max;
+      }
+
+      const pixelPerNumber = 6;
+      const minNumberNumbersToDisplay = 3;
+      const additionalPixelWidth =
+        (Math.round(maxValue).toString(10).length - minNumberNumbersToDisplay) * pixelPerNumber;
```

Comment:
Hm, I wonder, instead of trying to display numbers of very different lengths, wouldn't it be better to change from always displaying milliseconds to using human-readable values like 1.5s, 125ms, which we already do for trace latency in the search results?

Reply:
<img src="https://media0.giphy.com/media/duM6JZemPlOjUyqmxd/giphy.gif"/>

@afishler  WDYT?

Reply:
@yurishkuro I converted the numbers to another human readable time unit.
Please review my PR again 🙏